### PR TITLE
Implement public API bitmap support

### LINKED ISSUE #167: Allow storing bitmaps from any source.
### What
Currently, to support `Compose`, Shot provides a custom runner that captures allows to take those screenshots and the plugin picks them up. It would be useful to have a more generic approach such as
```kotlin
object ShotRunner {
  fun saveScreenshot(bitmap: Bitmap, name: String?)
}
```
One of the use-cases is to allow capturing screenshots using https://github.com/jraska/Falcon instead of the Facebook library. The improvement here is that it captures all windows allowing dialogs and popup windows to be captured as well.

### PR DESCRIPTION
**Pull request by pedrovgs**
### :pushpin: References
* **Issue:** Closes #167 

### :tophat: What is the goal?

Allow storing bitmaps from any source. 

### How is it being implemented?

We've created a new entry point in the ``ScreenshotTest`` API so we are able to use a Bitmap as part of the public API. Last but not least, we've added coverage using an end to end scenario 😃 

### How can it be tested?

🤖

### CODE REVIEW

**Review by flipper83 — APPROVED** [review]

File: shot-android/src/main/java/com/karumi/shot/compose/ComposeScreenshot.kt [hunk 1]
```diff
@@ -10,9 +11,14 @@ class ComposeScreenshot(
     private val saver: ScreenshotSaver
 ) {
 
+    fun saveScreenshot(bitmap: Bitmap, data: ScreenshotMetadata) {
```
Comment (Laimiux) [inline shot-android/src/main/java/com/karumi/shot/compose/ComposeScreenshot.kt:14]: Should this class be renamed and moved into a different package? I'm also wondering if this generic functionality for screenshots should be separated from `Compose` specific logic.

File: shot-android/src/main/java/com/karumi/shot/ScreenshotTest.kt [hunk 2]
```diff
@@ -112,6 +113,16 @@ interface ScreenshotTest {
         compareScreenshot(rule.onRoot(), name)
     }
 
+    fun compareScreenshot(bitmap: Bitmap, name: String? = null) {
```
Comment (Laimiux) [inline shot-android/src/main/java/com/karumi/shot/ScreenshotTest.kt:116]: Is this the public API we would call?

File: shot-android/src/main/java/com/karumi/shot/compose/ComposeScreenshot.kt (hunk 1 above)
Comment (pedrovgs) [reply shot-android/src/main/java/com/karumi/shot/compose/ComposeScreenshot.kt:14]: We can leave it as is for now. I don't want to extract it to another class when we don't know how this is going to evolve. As this is not part of the public API, it is ok if we just leave it here and we change it in any future refactor.

File: shot-android/src/main/java/com/karumi/shot/ScreenshotTest.kt (hunk 2 above)
Comment (pedrovgs) [reply shot-android/src/main/java/com/karumi/shot/ScreenshotTest.kt:116]: Yes 😃

**Pull request merged**